frontend/mobilityai/app/src/main/java/com/ai/mobility/mobilityai/MetaMotionDeviceAdapter.java

```diff
@@ -0,0 +1,183 @@
+        }
+
+        public void setDetails(MetaMotionDevice device) {
+            String signalString = "Signal Strength: ",
+                    macaddrString = "MAC Address: " + device.getMacAddr(),
+                    lastSynced ="Last Synced: " + device.getLastSync(),
+                    batteryLevel = String.format("Battery Level: %d", device.getBattery());
+            int rssi = device.getRssi();
+
+            //Signal Strength indicators - https://www.netspotapp.com/what-is-rssi-level.html
+            if(rssi > -50)
+                signalString += "Excellent";
+            else if(rssi > -60 && rssi <= -50)
+                signalString += "Good";
+            else if(rssi <= -60 && rssi > -70)
+                signalString += "Fair";
+            else if(rssi <= -70)
+                signalString += "Weak";
+            else
+                signalString += "Unknown";
+
+            devName.setText(device.getName());
+            devAssignedUser.setText(device.getAssignedUser());
+            devMacAddr.setText(macaddrString);
+            if(device.getAssignedUser().equals("Unassigned")) {
+                devMacAddr.setTextSize(15);
+                devMacAddr.setPadding(1,1,1,1);
+            } else {
+                devMacAddr.setTextSize(0);
+                devMacAddr.setPadding(0,0,0,0);
+            }
+
+            devBattery.setText(batteryLevel);
+            devLastSync.setText(lastSynced);
+            devSignalStrength.setText(signalString);
+            devProfileImage.setColorFilter(device.getColour());
+            devInitials.setText(device.getAssignedUser().substring(0, 1));
+        }
+
+        public void bind(MetaMotionDevice device, OnItemClickListener listener) {
+            itemView.setOnClickListener(new View.OnClickListener() {
+
+                @Override
+                public void onClick(View v) {
+                    listener.onItemClick(device);
+                }
+            });
+        }
+    }
+
+    public void clear() {
+        final int size = m_devices.size();
+        m_devices.clear();
+        notifyItemRangeRemoved(0, size);
+    }
+
+    public void update(MetaMotionDevice device) {
+        int pos = getPosition(device);
+
+        if(pos == -1) {
+            MetaMotionDevice dev = exists(device);
+            if(dev == null) {
+                device.setColour(getRandomMaterialColour());
+                m_devices.add(device);
+                notifyDataSetChanged();
+            } else
+            {
+                //Update rssi
+                dev.setRssi(device.getRssi());
+            }
+        }
+
+        Log.i("MobilityAI", "Added device, size = "+m_devices.size());
+
+
+        //TODO: Add RSSI stuff - https://github.com/mbientlab/BleToolbox/blob/master/scanner/src/main/java/com/mbientlab/bletoolbox/scanner/ScannedDeviceInfoAdapter.java
+
+        //TODO: Order by strength
+    }
+
+    private int getPosition(MetaMotionDevice device) {
```

Comment:
What is this for?

Reply:
It's for ordering the devices by signal strength, to be implemented later.

Reply:
#82